test/unit/range/views/view_kmer_hash_test.cpp

```diff
@@ -230,3 +230,19 @@ TYPED_TEST(kmer_hash_gapped_test, issue1754)
         EXPECT_RANGE_EQ(result_t{8}, text1 | stop_at_t | std::views::reverse | gapped_view);
     }
 }
+
+// https://github.com/seqan/seqan3/issues/1953
+TEST(kmer_hash_ungapped_test, issue1953)
+{
+    std::vector<seqan3::dna4> text1{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4}; // ACGTAGC
+    auto v = text1 | prefix_until_first_thymine | std::views::reverse | ungapped_view;
```

Comment:
```suggestion
    auto v = text1 | prefix_until_first_thymine | std::views::reverse | ungapped_view;
    auto const v_const = text1 | prefix_until_first_thymine | std::views::reverse | ungapped_view;
    EXPECT_EQ(1u, v_const.size());
```
Does this work?

Reply:
No, I get then the following error:

"error: passing ‘const seqan3::detail::kmer_hash_view<ranges::reverse_view<seqan3::detail::view_take_until<ranges::ref_view<std::vector<seqan3::dna4> >, <lambda(seqan3::dna4)>, false, false> > >’ as ‘this’ argument discards qualifiers [-fpermissive]
"

Is this unexpected?

Reply:
I already asked that: It doesn't.

Reply:
As it turns out it shouldn't work, because `std::views::reverse` drops const-iterability unless the view is a common range and this is not the case here. So it is expected that only the non-const version works.

Reply:
thanks for the error message & explanation you both :)